src/main/java/com/iota/iri/Iota.java

```diff
@@ -272,6 +280,24 @@ private void initializeTangle() {
         if (configuration.isZmqEnabled()) {
             tangle.addMessageQueueProvider(new ZmqMessageQueueProvider(configuration));
         }
+
+        if(configuration.isSelectivePermaEnabled()){
+            switch (configuration.getPermaMainDb()) {
+                case "rocksdb": {
+                    RocksDBPPPImpl ppp = new RocksDBPPPImpl(
+                            configuration.getPermaDbPath(),
+                            configuration.getPermaDbLogPath(),
+                            configuration.getPermaDbCacheSize());
+
+                    tangle.addPersistenceProvider(ppp);
+                    tangle.addPermanentPersistenceProvider(ppp);
+                    break;
+                }
+                default: {
+                    throw new NotImplementedException("No such database type.");
```

Comment:
I prefer to use `UnsupportedOperationException`, just because it doesn't have a dependency on a 3rd party library